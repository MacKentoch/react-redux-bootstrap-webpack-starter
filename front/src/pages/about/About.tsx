import React from 'react';
import { RouteComponentProps } from 'react-router';
import FadeInEntrance from '../../components/fadeInEntrance';
import { OwnProps, MappedStateToProps, MappedDispatchToProps } from './index';

// #region types
export type Props = RouteComponentProps &
  MappedDispatchToProps &
  MappedStateToProps &
  OwnProps;
// #endregion

function About(props: Props) {
  return (
    <FadeInEntrance>
      <h1>About</h1>
    </FadeInEntrance>
  );
}

About.displayName = 'About';

export default About;
