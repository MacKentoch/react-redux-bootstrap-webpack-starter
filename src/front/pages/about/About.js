// @flow

// #region imports
import React, { PureComponent } from 'react';
import {
  type Match,
  type Location,
  type RouterHistory,
} from 'react-router-dom';
// #endregion

// #region flow types
type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,

  ...any,
};
type State = any;
// #endregion

class About extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    return (
      <div>
        <h1>About</h1>
      </div>
    );
  }
  // #endregion
}

export default About;
