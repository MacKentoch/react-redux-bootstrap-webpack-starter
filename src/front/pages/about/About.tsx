import React from 'react';
import { RouteComponentProps } from 'react-router';
// import configureStore from 'redux-mock-store';
// import thunk from 'redux-thunk';
import FadeInEntrance from '../../components/fadeInEntrance';

// const middlewares = [thunk];
// const mockStore = configureStore(middlewares);

// #region types
type Props = {} & RouteComponentProps;
// #endregion

function About(props: Props) {
  return (
    <FadeInEntrance>
      <h1>About</h1>
    </FadeInEntrance>
  );
}

About.displayName = 'About';

export default About;
