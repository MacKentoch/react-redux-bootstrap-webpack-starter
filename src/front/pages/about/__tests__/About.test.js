// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import About from '../About';
// #endregion

describe('About page', () => {
  it('renders as expected', () => {
    const props = {
      actions: {},
    };
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <About {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
