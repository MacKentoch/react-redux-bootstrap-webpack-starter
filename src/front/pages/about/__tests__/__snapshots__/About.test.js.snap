// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page renders as expected 1`] = `
<div>
  <div>
    <h1>
      About
    </h1>
  </div>
</div>
`;
